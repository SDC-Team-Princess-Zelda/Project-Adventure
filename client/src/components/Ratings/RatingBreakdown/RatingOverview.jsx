import React from 'react';
import styled from 'styled-components';

import { useMeta } from '@Contexts/ReviewMeta';
import Star from '../../../Star';

export default function RatingOverview() {
  let score = 0;
  let helpfulness = null;
  const currentMeta = useMeta();

  if (currentMeta) {
    const { recommended, avgRating } = currentMeta;
    score = avgRating;

    // Handle helpfulness
    const trueCT = Number(recommended.true);
    const falseCT = Number(recommended.false);
    helpfulness = Math.round((trueCT / (trueCT + falseCT)) * 100);
  }

  return (
    <RatingOverviewContainer>
      <h3>
        RATINGS & REVIEWS
      </h3>
      <StarContainer>
        <Score>
          {score}
        </Score>
        <Star score={score} />
      </StarContainer>
      <div>
        {helpfulness}
        % of reviews recommend this product
      </div>
    </RatingOverviewContainer>
  );
}

// Style componnets
const RatingOverviewContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

const StarContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: row;
`;

const Score = styled.div`
  padding-right: 1rem;
  font-size: 4rem;
  font-weight: bold
`;
