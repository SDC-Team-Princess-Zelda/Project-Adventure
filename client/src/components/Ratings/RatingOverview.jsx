import React from 'react';
import styled from 'styled-components';
import { useTheme } from './ReviewContext';
import Star from '../../Star';

export default function RatingOverview() {
  let totalRating;
  let avgRating;
  let totalCT = 0;
  const currentMeta = useTheme();
  if (currentMeta) {
    const { ratings } = currentMeta;
    Object.keys(ratings).forEach((key) => {
      const currentCT = Number(ratings[key]);
      totalRating = key * currentCT;
      totalCT += currentCT;
    });
    avgRating = Math.round((totalRating / totalCT) * 10) / 10;
  }

  return (
    <RatingsSection>
      <h3>
        Ratngs & Reviews
      </h3>
      {avgRating}
      <Star />
      <div>
        TBD% of reivews recommend this product
      </div>
    </RatingsSection>
  );
}

const RatingsSection = styled.section`
  background-color: ${(props) => props.theme.colors.light};
`;
