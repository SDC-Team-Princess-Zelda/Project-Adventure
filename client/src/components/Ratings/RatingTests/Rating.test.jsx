/* eslint-disable no-undef */
import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import FactorEntry from '../RatingBreakdown/FactorEntry';

import RatingOverview from '../RatingBreakdown/RatingOverview';

// Snapshot Testing
it('RatingOverview renders correctly', () => {
  const tree = renderer
    .create(<RatingOverview />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// Rendering
let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders with factorBar descriptions', () => {
  act(() => {
    render(<FactorEntry factor="Size" />, container);
  });
  expect(container.textContent).toBe('Too smallPerfectToo large');

  act(() => {
    render(<FactorEntry factor="Comfort" />, container);
  });
  expect(container.textContent).toBe('PoorPerfect');
});

// Data Fetching
it("renders user data", async () => {
  const fakeUser = {
    name: "Joni Baez",
    age: "32",
    address: "123, Charming Avenue"
  };
  jest.spyOn(global, "fetch").mockImplementation(() =>
    Promise.resolve({
      json: () => Promise.resolve(fakeUser)
    })
  );