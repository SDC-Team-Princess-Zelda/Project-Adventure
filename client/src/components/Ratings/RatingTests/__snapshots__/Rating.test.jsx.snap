// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingOverview renders correctly 1`] = `
<div
  className="RatingOverview__RatingOverviewContainer-fsINFe kkezbv"
>
  <h3>
    RATINGS & REVIEWS
  </h3>
  <div
    className="RatingOverview__StarContainer-kUGGq dFxlKN"
  >
    <div
      className="RatingOverview__Score-iybahF iTAzDL"
    >
      0
    </div>
    <div
      className="Star__StarBlock-cVYQNc lhMbYs"
    >
      <div
        className="ratings"
      >
        <div
          className="empty-stars"
        />
        <div
          className="full-stars"
          style={
            Object {
              "width": "0%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="RatingOverview__RatingOverviewContainer-fsINFe kkezbv"
>
  <h3>
    RATINGS & REVIEWS
  </h3>
  <div
    className="RatingOverview__StarContainer-kUGGq dFxlKN"
  >
    <div
      className="RatingOverview__Score-iybahF iTAzDL"
    >
      0
    </div>
    <div
      className="Star__StarBlock-cVYQNc lhMbYs"
    >
      <div
        className="ratings"
      >
        <div
          className="empty-stars"
        />
        <div
          className="full-stars"
          style={
            Object {
              "width": "0%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
